dcos_installer/test_backend.py

```diff
@@ -67,6 +67,24 @@ def test_set_superuser_password(tmpdir):
         assert passlib.hash.sha512_crypt.verify('foo', config['superuser_password_hash'])
 
 
+def test_generate_node_upgrade_script(tmpdir):
+    """
+    Test the generate upgrade script works
+    """
+    config_extra = "\nmaster_list: ['10.0.0.1', '10.0.0.2', '10.0.0.5']\n"
+
+    with tmpdir.as_cwd():
+        genconf_dir = tmpdir.join('genconf')
+        genconf_dir.ensure(dir=True)
+        make_default_config_if_needed('genconf/config.yaml', config_extra)
+        genconf_dir.join('ip-detect').write('#!/bin/bash\necho 127.0.0.1')
+        tmpdir.join('artifacts/bootstrap/12345.bootstrap.tar.xz').write('contents_of_bootstrap', ensure=True)
+        tmpdir.join('artifacts/bootstrap/12345.active.json').write('{"active": "contents"}', ensure=True)
```

Comment:
I think you can replace these with the `create_config()` and `create_fake_build_artifacts()` functions from https://github.com/dcos/dcos/pull/1132. I think this unit test might actually fail without the artifacts helper, since after #1132 genconf will fail without the presence of a `complete.latest.json` build artifact.